src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -186,6 +187,15 @@ class ApplicationManagerImplTest
     app_manager_impl_->SetAppServiceManager(mock_app_service_manager_);
     Json::Value empty;
     ON_CALL(mock_last_state_, get_dictionary()).WillByDefault(ReturnRef(empty));
+
+    auto request = std::make_shared<smart_objects::SmartObject>(
+        smart_objects::SmartType_Map);
+    ON_CALL(*mock_message_helper_, CreateModuleInfoSO(_, _))
+        .WillByDefault(Return(request));
+    ON_CALL(*mock_policy_handler_, GetStatisticManager())
+        .WillByDefault(
+            Return(std::shared_ptr<usage_statistics::StatisticsManager>(
+                new usage_statistics_test::MockStatisticsManager())));
```

Comment:
@IGapchuk it's a bad practice. I suggest you to create this as a class member of `ApplicationManagerImplTest`:
```
usage_statistics_test::MockStatisticsManager mock_statistics_manager_;
...
ON_CALL(*mock_policy_handler_, GetStatisticManager()).WillByDefault(mock_statistics_manager_);
```

Or declare it as shared_ptr if you need exactly pointer.

Reply:
@AKalinich-Luxoft Done in0b50b09dccb77273e6047d277a99ef2ad28baaac